data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/model/DataPrepperConfiguration.java

```diff
@@ -144,6 +153,19 @@ private void setMetricTags(final Map<String, String> metricTags) {
         }
     }
 
+    public void setMetricTagFilters(final List<MetricTagFilter> metricTagFilters) {
+        if (metricTagFilters != null) {
+            metricTagFilters.forEach(
+                    metricTagFilter -> {
+                        if (metricTagFilter.getTags() != null && metricTagFilter.getTags().size() > MAX_TAGS_NUMBER) {
+                            throw new IllegalArgumentException("Tags cannot be more than 3 for each filter");
```

Comment:
Please re-word this phrase some. Here is one idea:

```
"Each metric tag filter may have no more than " + MAX_TAGS_NUMBER + " tags."
```

Please be sure to use the `MAX_TAGS_NUMBER` to avoid variations in the string and actual condition.